spacy/training/loggers.py

```diff
@@ -119,9 +124,23 @@ def setup_logger(
         for field in remove_config_values:
             del config_dot[field]
         config = util.dot_to_dict(config_dot)
-        wandb.init(project=project_name, config=config, reinit=True)
+        run = wandb.init(project=project_name, config=config, reinit=True)
         console_log_step, console_finalize = console(nlp, stdout, stderr)
 
+        def log_dir_artifact(
+            path: str,
+            name: str,
+            type: str,
+            metadata: Optional[Dict[str, Any]] = {},
+            aliases: Optional[List[str]] = [],
+        ):
+            dataset_artifact = wandb.Artifact(name, type=type, metadata=metadata)
+            dataset_artifact.add_dir(path, name=name)
+            wandb.log_artifact(dataset_artifact, aliases=aliases)
+
+        if log_dataset_dir:
+            log_dir_artifact(log_dataset_dir, "dataset", "dataset")
```

Comment:
Nitpicking comment: because there are several string arguments in `log_dir_artifact` and the order isn't intuitive to me, I'd personally call the function with named arguments for clarity, i.e.
```suggestion
            log_dir_artifact(path=log_dataset_dir, name="dataset", type="dataset")
```